llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -623,6 +670,12 @@ class Context {
   DenseMap<llvm::Value *, std::unique_ptr<sandboxir::Value>>
       LLVMValueToValueMap;
 
+  /// Remove \p V from the maps and returns the unique_ptr.
+  std::unique_ptr<Value> detachLLVMValue(llvm::Value *V);
```

Comment:
I'm debating whether or not this should have `[[nodiscard]]`. Unsure how pervasive you'd want `[[nodiscard]]` to be throughout these APIs

Reply:
I don't feel strongly either way, but these `detach()` functions are meant to be used either for detaching or for just deleting the value. I think it's currently only used for deleting them, so the value is discarded. There is no real harm in discarding the returned value as far as I can tell.

Reply:
I thought we wanted to keep around Values in case we rollback

Reply:
These APIs are for internal use only. The user won't be able to detach values. The only way a user can delete a value is through the `eraseFromParent()` API, and this will be tracked and can be rolled back.